examples/summarization/scripts/reward_summarization.py

```diff
@@ -0,0 +1,198 @@
+from datasets import load_dataset
+from transformers import (
+    AutoTokenizer,
+    AutoModelForSequenceClassification,
+    TrainingArguments,
+    Trainer,
+    PreTrainedTokenizerBase,
+    HfArgumentParser,
+)
+from transformers.utils import PaddingStrategy
+from typing import Optional, Union, List, Dict, Any
+import evaluate
+from dataclasses import dataclass, field
+import torch.nn as nn
+
+# Define and parse arguments.
+@dataclass
+class ScriptArguments:
+    """
+    These arguments vary depending on how many GPUs you have, what their capacity and features are, and what size model you want to train.
+    """
+
+    local_rank: Optional[int] = field(default=0, metadata={"help": "Used for multi-gpu"})
+    resume_from_checkpoint: Optional[bool] = field(
+        default=False, metadata={"help": "If you want to resume training where it left off."}
+    )
+    deepspeed: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Path to deepspeed config if using deepspeed. You may need this if the model that you want to train doesn't fit on a single GPU."
+        },
+    )
+    per_device_train_batch_size: Optional[int] = field(default=16)
+    per_device_eval_batch_size: Optional[int] = field(default=16)
+    gradient_accumulation_steps: Optional[int] = field(default=4)
+    learning_rate: Optional[int] = field(default=2e-5)
+    model_name: Optional[str] = field(
+        default="gpt2",
+        metadata={
+            "help": "The model that you want to train from the Hugging Face hub. E.g. gpt2, gpt2-xl, bert, etc."
+        },
+    )
+    bf16: Optional[bool] = field(
+        default=False,
+        metadata={
+            "help": "This essentially cuts the training time in half if you want to sacrifice a little precision and have a supported GPU."
+        },
+    )
+
+
+parser = HfArgumentParser(ScriptArguments)
+script_args = parser.parse_args_into_dataclasses()[0]
+
+# Load the human comparisons dataset for tuning the reward model.
+ds = load_dataset("openai/summarize_from_feedback", name="comparisons")
+
+# Define the training args. Needs to be done before the model is loaded if you are using deepspeed.
+training_args = TrainingArguments(
+    output_dir=f"{script_args.model_name}_summarization_reward_model",
+    learning_rate=script_args.learning_rate,
+    per_device_train_batch_size=script_args.per_device_train_batch_size,
+    per_device_eval_batch_size=script_args.per_device_eval_batch_size,
+    num_train_epochs=5,
```

Comment:
That seems like a parameter someone might also want to tune, right? so maybe move it to the arguments as well.

Reply:
Yeah really good point. 5 is the number of epochs that openai used, but it makes sense for me to make it an option and just make 5 the default. And then put that in the documentation.